src/Network/WebSockets/TSSocketManager.m

```diff
@@ -211,7 +221,13 @@ - (void)setStatus:(SocketStatus)status {
             socket.delegate = self;
             
             [self setWebsocket:socket];
+
+            // [SRWebSocket open] could hypothetically call a delegate method (e.g. if
+            // the socket failed immediately for some reason), so we update the status
+            // _before_ calling it, not after.
+            _status = status;
             [socket open];
+            return;
```

Comment:
I don't think this is necessary in practice, but this isn't the first time we've hit edge cases with our web socket, so I'm trying to be super defensive.